src/controller.rs

```diff
@@ -217,21 +234,168 @@ async fn reconcile(app_instance: Arc<AppInstance>, ctx: Arc<Context>) -> Result<
+        } else {
+            Ok(Action::await_change())
+        }
+    }
+}
+
+async fn launch_cleanup_job(app_instance: &AppInstance, ctx: &Context) -> Result<()> {
+    let ns = &app_instance.namespace().ok_or(Error::NamespaceRequired)?;
+    let cleanup_job_name = format!("kubit-cleanup-{}", &app_instance.name_any());
+
+    let mut volumes = vec![Volume {
+        name: "manifests".to_string(),
+        empty_dir: Some(Default::default()),
+        ..Default::default()
+    }];
+
+    if let Some(ref refs) = app_instance.spec.image_pull_secrets {
+        let secret_ref = refs
+            .iter()
+            .exactly_one()
+            .map_err(|_| Error::UnsupportedMultipleImagePullSecrets)?;
+
+        let docker_creds = Volume {
+            name: "docker".to_string(),
+            secret: Some(SecretVolumeSource {
+                secret_name: secret_ref.name.clone(),
+                items: Some(vec![KeyToPath {
+                    key: ".dockerconfigjson".to_string(),
+                    path: "config.json".to_string(),
+                    ..Default::default()
+                }]),
+                ..Default::default()
+            }),
+            ..Default::default()
+        };
+        volumes.push(docker_creds);
+    }
+
+    let mk_mount = |name: &str| VolumeMount {
+        name: name.to_string(),
+        mount_path: format!("/{name}"),
+        ..Default::default()
+    };
+
+    let volume_mounts = Some(volumes.iter().map(|v| mk_mount(&v.name)).collect());
+    let container_defaults = Container {
+        volume_mounts: volume_mounts.clone(),
+        env: Some(vec![
+            EnvVar {
+                name: "DOCKER_CONFIG".to_string(),
+                value: Some("/docker".to_string()),
+                ..Default::default()
+            },
+            EnvVar {
+                name: "KUBECTL_APPLYSET".to_string(),
+                value: Some("true".to_string()),
+                ..Default::default()
+            },
+        ]),
+        ..Default::default()
+    };
+
+    let jobs: Api<Job> = Api::namespaced(ctx.client.clone(), ns);
+    let job = Job {
+        metadata: ObjectMeta {
+            name: Some(cleanup_job_name),
+            namespace: app_instance.namespace().clone(),
+            owner_references: owned_by(app_instance),
+            ..Default::default()
+        },
+        spec: Some(JobSpec {
+            backoff_limit: Some(0),
+            template: PodTemplateSpec {
+                spec: Some(PodSpec {
+                    service_account: Some(APPLIER_SERVICE_ACCOUNT.to_string()),
+                    restart_policy: Some("Never".to_string()),
+                    active_deadline_seconds: Some(180),
+                    volumes: Some(volumes),
+                    init_containers: Some(vec![Container {
+                        name: "setup-delete".to_string(),
+                        image: Some("index.docker.io/jdockerty/kubit-local:latest".to_string()), // TODO: update on release
```

Comment:
Drawing special attention to this, it'll be changed to `ctx.kubit_image` once we have a new release.

For now, the `kubit` release does not have the `kubit helper cleanup` subcommand, so I had to build my own to get it to pass tests.